src/modules/lifecycle.ts

```diff
@@ -1,21 +1,21 @@
 /**
  * Utilities for monitoring and updating the lifecycle of a PureCloud Client App
  *
- * @module modules/lifecycle
- *
  * @since 1.0.0
  */
 
-import BaseApi from './base';
+import BaseApi, { SDKMessagePayload } from './base';
+
+export type LifecycleHook = 'bootstrap' | 'focus' | 'blur' | 'stop';
 
 const LIFECYCLE_HOOK_EVENT_NAME = 'appLifecycleHook';
-const HOOK_NAME_FILTER = function (hookName, msgPayload) {
+const buildHookFilter = (hookName: LifecycleHook) => (msgPayload: SDKMessagePayload) => {
```

Comment:
Any pro/con to importing MessagePayloadFilter and using that type as the return type for this function?

Reply:
Not opposed here, it really comes down to a slight change in the function signature. Typescript will enforce that these 2 return types to be in sync since one eventually feeds down into the other, but I do like importing since the only reason this guy even exists is to eventually feed down into the listener options.